cmake/CMakeLists.txt

```diff
@@ -60,8 +60,11 @@ endif()
 option(COLVARS_TCL "Link against the Tcl library" OFF)
 
 if(COLVARS_TCL)
-  find_library(TCL_LIBRARY NAMES tcl PATHS HINTS "${TCL_DIR}" PATH_SUFFIXES lib lib64)
-  target_include_directories(colvars_obj PRIVATE ${TCL_DIR}/include)
+  find_package(TCL)
```

Comment:
The workflow for building TinkerHP involves building the static library, so I'd like that to be as simple as possible, that is, to happen by default. The new situation is, these builds are not "just for tests" anymore, which is good news.
Using an object library, building the shared and static library at once is fairly lightweight. But it could also be controlled by the `BUILD_SHARED_LIBS` option, that's fine with me.

Reply:
Ok, in that case I suggest:
1) Remove or modify the message about the CMake recipe being only for tests.
2) Add an entry in `update-colvars-code.sh` to patch TinkerHP's Makefile and print out a suggested CMake command line.

Reply:
> 1. Remove or modify the message about the CMake recipe being only for tests.
Agreed!

> 2. Add an entry in `update-colvars-code.sh` to patch TinkerHP's Makefile and print out a suggested CMake command line.

That's definitely going to happen, but the TinkerHP Makefile for Colvars is not completely stable yet, so this depends on a little cleanup work first. I wouldn't mind merging this without waiting for that, because those future changes will be mostly orthogonal to the work happening in master.